test/es2015/proxy-propertyKey-typeof.js

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2019-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// dummy call, to invoke proxy's get
+function dummyCall() {
+}
+
+
+// check if string conversion is done
+class StrArray extends Array {
+    constructor(){
+        super(...arguments);
+        this.index = {}
+        return new Proxy(this, this.handler);
+    }
+    get handler () {
+        let self = this;
+        return {
+            get(target, prop, proxy){
+               assert(typeof prop == "string")
```

Comment:
For example, we can just allocate a proxy object and its simple handler object to check if the type of property is string or not, similar to the below code that checks the symbol type.

Reply:
That's a good idea, i updated the PR with a similar check as with the Symbol.